sqlmesh/core/model/definition.py

```diff
@@ -1888,6 +1925,20 @@ def _python_env(
                         used_macros[name] = macros[name]
                     elif name in variables:
                         used_variables.add(name)
+                elif (
+                    macro_func_or_var.__class__ is exp.Identifier and "@" in macro_func_or_var.this
+                ):
+                    var_name = None
+                    macro_template = macro_func_or_var.this
+                    at_idx = macro_template.index("@")
```

Comment:
what if there are two of them? are you sure you don't wanna use regex?

```
x_@{y}_@{z}

Reply:
ah, good point. Wanted to save on regex

Reply:
Turned out I could just use the pattern compiled as part of the template class construction. Fixed this